src/idl_gen_text.cpp

```diff
@@ -51,11 +51,24 @@ bool Print(T val, Type type, int /*indent*/, Type * /*union_type*/,
            const IDLOptions &opts, std::string *_text) {
   std::string &text = *_text;
   if (type.enum_def && opts.output_enum_identifiers) {
-    auto ev = type.enum_def->ReverseLookup(static_cast<int64_t>(val));
-    if (ev) {
-      text += "\"";
-      text += ev->name;
-      text += "\"";
+    std::vector<EnumVal const *> enum_values;
+    if (auto ev = type.enum_def->ReverseLookup(static_cast<int64_t>(val))) {
+      enum_values.push_back(ev);
+    }
+    else if (val && type.enum_def->attributes.Lookup("bit_flags")) {
+      for (auto it = type.enum_def->Vals().begin(),
+                e = type.enum_def->Vals().end();
+           it != e; ++it) {
+        if ((*it)->GetAsInt64() & static_cast<int64_t>(val))
```

Comment:
`uint64_t` will be better here to avoid overflow (UBSAN) if `T` of `val` is uint64_t.
By the grammar, `bit_flag` enums should have unsigned underlying type.

Reply:
good point